src/main/java/org/apache/sling/graphql/core/engine/DefaultQueryExecutor.java

```diff
@@ -394,6 +417,24 @@ TypeDefinitionRegistry getTypeDefinitionRegistry(@NotNull String sdl, @NotNull R
         }
     }
 
+    private Consumer<GraphQLContext.Builder> getGraphQLContextBuilder() {
+        ParserOptions defaultParserOptions = ParserOptions.getDefaultParserOptions();
+        Consumer<GraphQLContext.Builder> graphQLContextBuilder = builder -> {
```

Comment:
Can we use [ParserOptions#transform](https://javadoc.io/static/com.graphql-java/graphql-java/20.1/graphql/parser/ParserOptions.html#transform-java.util.function.Consumer-) here, for simpler code?

IIUC that'd look something like

    ParserOptions opt = ParserOptions.getDefaultParserOptions().transform(builder -> {
      builder
      .maxTokens(queryMaxTokens)
      .maxWhitespaceTokens(queryMaxWhitespaceTokens)
      .build();
    }
    
...untested code, just looking at the docs!

Reply:
Pushed something like the above, works.